### PR TITLE
jsonized EP clode clean up

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary

```SUMMARY: None```

#### Purpose of change
Applies @jbytheway suggestion from #32939 

#### Describe the solution
Use map property properly, remove useless special case.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -10189,10 +10189,8 @@ int player::get_env_resist( body_part bp ) const
     }
 
     for( const bionic &bio : *my_bionics ) {
-        for( const auto &element : bio.info().env_protec ) {
-            if( element.first == bp || ( bp == bp_eyes && element.first == bp_head ) ) {
-                ret += element.second;
-            }
+        if( bio.info().env_protec.find( bp ) != bio.info().env_protec.cend() ) {
```
Comment (jbytheway) [inline src/player.cpp]:
```suggestion
        if( bio.info().env_protec.find( bp ) != bio.info().env_protec.end() ) {
```
Comparing to `cend` will work, but it's an unconventional choice.

File: src/player.cpp [hunk 2]
```diff
@@ -10189,10 +10189,8 @@ int player::get_env_resist( body_part bp ) const
...
+            ret += bio.info().env_protec.at( bp );
```
Comment (jbytheway) [inline src/player.cpp]: If you assign the result from `find` to a variable you won't need to perform a second lookup here.

Comment (Fris0uman) [reply src/player.cpp]: Better now?

Comment (jbytheway) [reply src/player.cpp]: Yep :).

**Pull request merged**